pkg/filemanager/filemanager.go

```diff
@@ -44,13 +46,91 @@ func NewFileManager(path string) (FileManager, error) {
 		return NewGCSStorageFile(path)
 	case strings.Contains(path, "blob.core.windows.net"):
 		return NewAzureBlobFile(path)
+	case strings.HasPrefix(path, "http://"), strings.HasPrefix(path, "https://"):
```

Comment:
With this change `s3` protocol will be assumed for any `http(s)` link.
It will require a breaking change for a potential extension in the future.

I would suggest instead using one of the 3 next approaches:
1) Introduce new protocol string, like `customs3://endpoint.com/my-file.csv` or `minio://endpoint.com/my-file.csv`
2) Add domain filter like, minio.com (probably the least flexible option)
3) Extend existing `S3File` struct to support new destination. (I'm not familiar with minio, but it looks like you're using existing aws library and the same method calls.)

Reply:
I'm leaning towards option 1 because I'd rather don't want to mess up the current AWS implementation too much or add extra complexity to it. I would like to hear your thoughts on the following. Introduce a new protocol prefix string sort of like you mentioned, e.g. `customs3://fqdn:port/bucket-name/path/to/file.csv` and always enforce secure https so this translates to: `https://fqdn:port/bucket-name/path/to/file.csv`.

Reply:
Sounds good to me. 

It also seems like any`http` vs `https` selection happens inside AWS SDK.

Reply:
I changed the protocol prefix string/scheme to `alts3` (alternative S3) so `alts3://fqdn:port/bucket-name/path/to/file.csv` translates to `https://fqdn:port/bucket-name/path/to/file.csv`.